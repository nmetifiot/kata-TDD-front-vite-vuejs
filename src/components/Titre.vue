<script setup></script>

<template>
  <h1>Kata TDD</h1>

</template>
