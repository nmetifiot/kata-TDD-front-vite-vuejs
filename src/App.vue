<script setup>
import Titre from "./components/titre.vue";
import Joke from "./components/joke.vue";
</script>

<template>
<Titre />
  <Joke />
</template>
