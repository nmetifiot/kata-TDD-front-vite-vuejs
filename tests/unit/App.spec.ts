import {describe, expect, it} from "vitest";
import {mount} from "@vue/test-utils";
import App from "@/App.vue";
import Titre from "@/components/Titre.vue";
import Joke from "@/components/joke.vue";

const generateWrapper = () => mount(App, {shallow: true});

describe('App', () => {
    describe('render', () => {
        it('should render correctly', () => {
            const wrapper = generateWrapper();

            expect(wrapper.html()).toMatchSnapshot();
        });
        it('should have title', () => {
            const wrapper = generateWrapper()

            const titleComponent = wrapper.findComponent(Titre);

            expect(titleComponent.exists()).toBe(true);
        });
        it('should have component Joke', () => {
            const wrapper = generateWrapper()

            const titleComponent = wrapper.findComponent(Joke);

            expect(titleComponent.exists()).toBe(true);
        })
    })
})