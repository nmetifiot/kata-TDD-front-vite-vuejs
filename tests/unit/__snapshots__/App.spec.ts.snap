// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`App > render > should render correctly 1`] = `"<h1>Kata TDD</h1>"`;
